python/lsst/dax/imgserv/getimage/imageget.py

```
@@ -0,0 +1,625 @@
+        metadata = self._butler.get(self._imagedataset_md(), dataId=data_id)
+        return metadata
+
+    def _image_from_butler(self, data_id, bbox=None):
+        # Retrieve the image through the Butler using data id.
+        self._log.debug("_image_from_butler data_id:{}".format(data_id))
+        if bbox:
+            image = self._butler.get(self._imagedataset_sub(), bbox=bbox,
+                                     dataId=data_id, immediate=True)
+        else:
+            image = self._butler.get(self._imagedataset_type, dataId=data_id)
+        return image
+
+    def _imagedataset_md(self):
+        # Return the butler policy name to retrieve metadata
+        return self._imagedataset_type + "_md"
+
+    def _imagedataset_sub(self):
+        # Return the dataset type for sub-images
+        return self._imagedataset_type + "_sub"
+
+
+    def _cutout_from_src(self, src_image, xy_center_x, xy_center_y, width,
+                         height, wcs):
+        # Returns an image cutout from the source image.
+        # srcImage - Source image.
+        # xy_center - The center of region to cutout in pixels.
+        # width - The width in pixels.
+        # height - The height in pixels.
+        # height and width trimmed if they go past the edge of source image.
+        # First, center the cutout image.
+        pix_ulx = int(xy_center_x - width / 2.0)
+        pix_uly = int(xy_center_y - height / 2.0)
+        xy_center = afwGeom.Point2I(pix_ulx, pix_uly)
+        log.debug("xy_center={}".format(xy_center))
+        src_box = src_image.getBBox()
+        # assuming both src_box and xy_center to be in Box2I
+        co_box = afwGeom.Box2I(xy_center,
+                               afwGeom.Extent2I(int(width), int(height)))
+        if co_box.overlaps(src_box):
+            co_box.clip(src_box)
+        else:
+            self._log.debug(
+                "cutout image wanted is OUTSIDE source image -> None")
+            return None
```

Throw an exception instead of returning None?